src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumChest.java

```diff
@@ -178,6 +198,37 @@ public void readFromNBT(NBTTagCompound data) {
         }
     }
 
+    @Override
+    public void initFromItemStackData(NBTTagCompound itemStack) {
+        super.initFromItemStackData(itemStack);
+        if (itemStack.hasKey("ItemStack", NBT.TAG_COMPOUND)) {
+            this.itemStack = new ItemStack(itemStack.getCompoundTag("ItemStack"));
+            if (!this.itemStack.isEmpty()) {
+                this.itemsStoredInside = itemStack.getLong("ItemAmount");
+            }
+        } else if (itemStack.hasKey("PartialStack", NBT.TAG_COMPOUND)) {
+            importItems.setStackInSlot(0, new ItemStack(itemStack.getCompoundTag("PartialStack")));
+        }
+    }
+
+    @Override
+    public void writeItemStackData(NBTTagCompound itemStack) {
```

Comment:
There is problem with calculating number of items on stacks when full stack is not 64 items. As on pictures below I have put in 2 full stacks (but capacity is 16) and got back 5 full stacks.

![2021-03-31_17 02 54](https://user-images.githubusercontent.com/3093101/113167141-b90c7e80-9243-11eb-9551-af078455954a.png)
![2021-03-31_17 03 01](https://user-images.githubusercontent.com/3093101/113167147-ba3dab80-9243-11eb-88ac-61636b8dea4a.png)